sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/ClientRetryPolicy.java

```diff
@@ -372,8 +379,12 @@ private Mono<ShouldRetryResult> shouldRetryOnBackendServiceUnavailableAsync(
         CosmosException cosmosException) {
 
         if (this.globalPartitionEndpointManagerForPerPartitionCircuitBreaker.isPerPartitionLevelCircuitBreakingApplicable(this.request)) {
-            this.globalPartitionEndpointManagerForPerPartitionCircuitBreaker
-                .handleLocationExceptionForPartitionKeyRange(this.request, this.request.requestContext.regionalRoutingContextToRoute);
+            try {
+                this.globalPartitionEndpointManagerForPerPartitionCircuitBreaker
+                    .handleLocationExceptionForPartitionKeyRange(this.request, this.request.requestContext.regionalRoutingContextToRoute, false);
+            } catch (CosmosException e) {
+                logger.warn("Per-partition circuit breaker hit an exception ", e);
```

Comment:
ditto

Reply:
Fixed in commit - https://github.com/Azure/azure-sdk-for-java/pull/46968/commits/862e4e3e1c3062ad0a67aff047fe4a7d54219f80